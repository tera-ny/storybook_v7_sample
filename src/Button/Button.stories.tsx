import { Meta, StoryObj } from "@storybook/react"
import { Button } from "."

export default {
  component: Button,
} as Meta

export const Default: StoryObj<typeof Button> = {};