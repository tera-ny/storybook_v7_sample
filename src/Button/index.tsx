import { FC } from "react";
import { buttonStyle } from "./module.css"

export const Button: FC = () => <button className={buttonStyle}>button</button>