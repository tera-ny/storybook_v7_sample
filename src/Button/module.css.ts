import {style} from "@vanilla-extract/css"

export const buttonStyle = style({
  fontSize: "24px"
})