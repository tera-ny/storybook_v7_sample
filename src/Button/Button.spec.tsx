import { composeStory } from "@storybook/testing-react";
import { render } from "@testing-library/react";
import * as stories from "./Button.stories";

const cases: [stories: keyof typeof stories][] = [["Default"]];

describe("Button component", () => {
  describe.each(cases)("%s", (story) => {
    const Story = composeStory(stories[story], stories.default);

    it("render component", () => {
      const { container } = render(<Story />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
