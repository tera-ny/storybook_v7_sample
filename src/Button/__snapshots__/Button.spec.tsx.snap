// Vitest Snapshot v1

exports[`Button component > Default > render component 1`] = `
<button
  class="module_buttonStyle__illoh50"
>
  button
</button>
`;
